### ISSUE TITLE
Consider unvendoring joblib

### ISSUE THREAD

**Issue by rth** [conda-forge/scikit-learn-feedstock#75]
Currently joblib comes vendored inside scikit-learn. In the context of conda there is no reason to do that, and with https://github.com/scikit-learn/scikit-learn/pull/11471 it is possible to use a site joblib. 

So one could imagine adding joblib as an explicit dependency and possibly removing `sklearn.externals.joblib` (or at least patching it to raise a deprecation warning).

Though I'm not sure if this has a potential to break compatibility with existing pickles (hopefully running tests might be enough to detect that?).

The advantage is that bug fixes from joblib/loky will become independent from scikit-learn releases. 

If this is accepted, getting some feedback from conda-forge users could help with unvendoring of joblib in the long term.

**Comment by amueller** [comment]
In which context is there a reason for it?

**Comment by rth** [comment]
I would be all for unvendoring in general, and I agree it's worth discussing at scikit-learn, but from what I understood there may not be full consensus on this subject yet and this pragmatically aims to make it happen on distribution channels where it's less controversial as a first step.

**Comment by amueller** [comment]
Doing it differently in different distribution channels seems like it might cause more issues, though?
I'm kinda waiting on the governance doc on this one, which will allow us to make decisions without consensus.

**Comment by rth** [comment]
> Doing it differently in different distribution channels seems like it might cause more issues, though?

Yeah, that's a possible risk. But we are officially supporting non vendored joblib installations with [`SKLEARN_SITE_JOBLIB`](http://scikit-learn.org/stable/modules/computing.html#environment-variables) and if there are issues with it it would be good to know. Also this use-case is now [tested in CI](https://github.com/scikit-learn/scikit-learn/blob/5df8cd3bd7fe709325e385b20df51f5c096774ae/.travis.yml#L49). This would be merely increasing the usage of this use case with respect to the one where the the vendored version is used, not really introducing anything new. 

> I'm kinda waiting on the governance doc on this one,

I agree that having a clearly defied governance model would be good in any case.

**Comment by rth** [comment]
Also I think some Linux distributions may be unvendoring already or at least [trying to](https://github.com/gentoo/gentoo/blob/73b724093b9c2a8756b8c35d3e09793342fa9ca9/sci-libs/scikits_learn/scikits_learn-0.19.0.ebuild#L60).

**Comment by rth** [comment]
This will take place as soon as joblib is unvendored in the scikit-learn (https://github.com/scikit-learn/scikit-learn/pull/13531). Closing.
